### ISSUE TITLE
[WIP] Things to do 

### ISSUE THREAD

**Issue by TylorS** [TylorS/tempest#1]
This is a general list of things that need to be done before anything is really even near completion

REMINDER: Update docs as I go or be miserable rewriting them at the end of it all
## Immediate needs
- [x] Simplify Stream types to be identified by their Source
- [x] Only have 1 Stream type in `core` that is Multicast Only
- [x] remove `unshare()` `remember()` `share()` from prototypes
## Operators
- [x] mapTo - should this be in `map` ?
- [x] filter
- [x] take
- [x] drop
- [x] last
- [x] startWith
- [x] endWhen
- [x] fold
- [x] replaceError
- [x] flatten
- [x] remember - should be home to `MemoryStream` type
- [x] debug - Extra ideas for the future
  - [ ] Wild idea: if available, use ES6 Proxies to inspect the source chain and provide useful info
  - [ ] Provide info on when a subscriber (un)subscribes, receives events etc
- [x] proxy - don't want to use the name imitate
- [x] sample - sampling is very common, and in combination with combine() makes withLatestFrom
## Factories
- [x] create
- [x] never
- [x] empty
- [x] throw
- [x] fromPromise - maybe go in core to be used in `from()` ?
- [x] periodic
- [x] merge
- [x] combine
## Extra stuff
- [x] optimize Multicast
- [x] curry all the things
- [x] Investigate Subject implementation 
  - Probably could easily 'upgrade' a stream
  - Proof of Concept http://www.webpackbin.com/N1_ozN_DW
- [ ] Add performance tests
- [ ] Add documentation generation 
- [x] Investigate [builder](https://github.com/FormidableLabs/builder) to make things DRY
- [ ] Document/~~Simplify~~ how to create a new package
- [ ] Build Cycle.js adapter and run()

**Comment by TylorS** [comment]
Closing in favor of #7
